tests/ui/UnreadIndicatorsTest.js

```diff
@@ -0,0 +1,395 @@
+import ONYXKEYS from '../../src/ONYXKEYS';
+import waitForPromisesToResolve from '../utils/waitForPromisesToResolve';
+import * as TestHelper from '../utils/TestHelper';
+import appSetup from '../../src/setup';
+import fontWeightBold from '../../src/styles/fontWeight/bold';
+import * as AppActions from '../../src/libs/actions/App';
+import ReportHeaderViewBackButton from '../../src/pages/home/ReportHeaderViewBackButton';
+import ReportActionsView from '../../src/pages/home/report/ReportActionsView';
+import * as NumberUtils from '../../src/libs/NumberUtils';
+import LocalNotification from '../../src/libs/Notification/LocalNotification';
+import * as Report from '../../src/libs/actions/Report';
+
+beforeAll(() => {
+    global.fetch = TestHelper.getGlobalFetchMock();
+
+    // We need a bit more time for this test in some places
+    jest.setTimeout(30000);
+    Linking.setInitialURL('https://new.expensify.com/r/1');
+    appSetup();
+});
+
+/**
+ * @param {RenderAPI} renderedApp
+ */
+function scrollUpToRevealNewMessagesBadge(renderedApp) {
+    fireEvent.scroll(renderedApp.getByTestId('report-actions-list'), {
+        nativeEvent: {
+            contentOffset: {
+                y: 250,
+            },
+            contentSize: {
+                // Dimensions of the scrollable content
+                height: 500,
+                width: 100,
+            },
+            layoutMeasurement: {
+                // Dimensions of the device
+                height: 700,
+                width: 300,
+            },
+        },
+    });
+
+    // We advance the timer since we must wait for the animation to end
+    // and the new style to be reflected
+    jest.advanceTimersByTime(100);
+}
+
+/**
+ * @param {RenderAPI} renderedApp
+ * @return {Boolean}
+ */
+function isNewMessagesBadgeVisible(renderedApp) {
+    const badge = renderedApp.getByTestId('new-messages-badge');
+    return badge.props.style.transform[0].translateY === 10;
+}
+
+/**
+ * @param {RenderAPI} renderedApp
+ * @return {Promise}
+ */
+async function navigateToSidebar(renderedApp) {
+    const reportHeader = renderedApp.getByTestId('report-header');
+    const reportHeaderBackButton = reportHeader.findByType(ReportHeaderViewBackButton);
+    fireEvent(reportHeaderBackButton, 'press');
+    return waitForPromisesToResolve();
+}
+
+/**
+ * @param {RenderAPI} renderedApp
+ * @param {Number} index
+ * @return {Promise}
+ */
+async function navigateToSidebarOption(renderedApp, index) {
+    const optionRows = renderedApp.getAllByTestId('option-row');
+    fireEvent(optionRows[index], 'press');
+    await waitForPromisesToResolve();
+}
+
+/**
+ * @param {RenderAPI} renderedApp
+ * @return {Boolean}
+ */
+function isDrawerOpen(renderedApp) {
+    const reportScreen = renderedApp.getByTestId('report-screen');
+    return reportScreen.findByType(ReportActionsView).props.isDrawerOpen;
+}
+
+test('Unread LHN Status, “New Messages” badge, and New Line Indicator', async () => {
```

Comment:
Using `describe()` and `it()` will make it consistent with our other tests

Reply:
Do we have a policy written down which says it's OK to use async/await in unit tests?

Reply:
>Do we have a policy written down which says it's OK to use async/await in unit tests?

I can add something. 99% sure these these tests will not work without it and if they do will be much less readable 😄

Reply:
Haha touché and very understandable. I'm also seeing a different kind of value in just following the doc examples and using the testing library as it was intended to be used. Most if not all examples I've come across while learning the testing library use async/await. So while I understand there is value in consistency I don't completely love that we will chose to not follow those examples. It feels counter-intuitive to learn something one way and then take an extra step to decompile it all into something else.

Reply:
After looking at the tests for a bit I was able to make this 99% async/await free 🎉 

Unfortunately, there is still a single function that is using async/await, but I moved it out of the test file and put some disclaimers around it. From what I can gather, there is some magic happening with the [`act()` method](https://reactjs.org/docs/test-utils.html#act) that waits for state updates to occur in React components, but for some reason I cannot get this one warning to go away without it.

>Warning: You called act(async () => ...) without await. This could lead to unexpected testing behaviour, interleaving multiple act calls and mixing their scopes. You should - await act(async () => ...);

It _only_ goes away if I use async/await. Curious if you have a better understanding than me about how the `act()` method works and whether you'd be comfortable with us using async/await for this one case?